docs/spec/concepts.rst

```diff
@@ -3,6 +3,268 @@
 Type system concepts
 ====================
 
+Static, dynamic, and gradual typing
+-----------------------------------
+
+A **statically-typed** programming language runs a type-checker before running
+a program. The program is required to be well-typed according to the language's
+type system. The type system assigns a type to all expressions in the language
+and verifies that their uses obey the typing rules. Normally, a program that is
+not well-typed (i.e., one that contains a type error) will not run. Java and
+C++ are examples of statically-typed object-oriented languages.
+
+A **dynamically-typed** programming language does not run a type-checker before
+running a program. Instead, it checks the types of values before performing
+operations on them at runtime. This is not to say that the language is
+"untyped". Values at runtime have a type and their uses obey typing rules. Not
+every operation will be checked, but certain primitive operations in the
+language such as attribute access or arithmetic are. Python is a
+dynamically-typed language.
+
+**Gradual typing** is a way to combine static and dynamic typing.
+Type-annotated Python allows opting in to static type checking at a fine level
+of granularity, so that some type errors can be caught statically, before
+running the program. Variables, parameters, and returns can optionally be given
+a static type annotation. Even within the type of a single data structure,
+static type checking is opt-in and granular. For example, a dictionary can be
+annotated to have static checking of the key type but only dynamic checking of
```

Comment:
Taking the first suggestion ("enable" instead of "have") but not the second: to me "retain" wrongly suggests that otherwise we'd be giving up dynamic checking, which isn't actually the case in Python (as you highlight in a later comment.)

Reply:
I see, makes sense. I think I interpreted "dynamic typing" here to refer to the way the static type checker was meant to see the types (i.e., they should interpret anything unannotated as being of type `Any`). Perhaps something like this might be clearer:

```suggestion
annotated to enable static checking of the key type but only have dynamic runtime checking of
```